### PR TITLE
Converter bug fixes

### LINKED ISSUE #9698: Converters: generic converter is not instantiated when multiple converters are involved
This one is easier to describe from compiler internals perspective.
When within one call multiple converters are used, for example first argument is converted using generic converter and second argument is converted using non generic converter then flag ``genericConverter`` in ``TCandidate`` gets overridden by second converter to ``false`` and instantiation of first converter is not going to happen

I will provide a fix.

### LINKED ISSUE #9699: Converters returning var T or lent T produce broken c code
For converters returning ``var T`` or ``lent T`` generated CPP does not compile.
For C backend, generated C compiles but produces garbage output at the runtime.

I will provide a fix

### PR DESCRIPTION
**Pull request by cooldome**
This change  paves the way for converters that don't copy the objects by using `lent T`
Test case is included.

Fixes #9699
Fixes #9698

### CODE REVIEW

**Comment by cooldome** [comment]
I have an idea for different approach this fix. 
Existing fix: userConvMatch adds hiddenDeref node when converter returns ``var T``  and then ``analyseIfAddressTakenInCall`` removes it if it is not required by calling ``newHiddenAddrTaken``.

New idea: ``analyseIfAddressTakenInCall``  will add ``hiddenDeref`` on top of converter calls when required

Let me know which one is preferred.

**Comment by cooldome** [comment]
Could you please review as a priority.
? this a blocker for me.

File: compiler/sigmatch.nim [hunk 1]
```diff
@@ -1855,8 +1855,14 @@ proc userConvMatch(c: PContext, m: var TCandidate, f, a: PType,
...
         param = copyTree(arg)
       addSon(result, param)
+
+      if dest.kind in {tyVar, tyLent}:
+        dest.flags.incl tfVarIsPtr
+        result = newDeref(result)
+
       inc(m.convMatches)
-      m.genericConverter = srca == isGeneric or destIsGeneric
+      if m.genericConverter == false:
```
Comment (Araq) [inline compiler/sigmatch.nim]: ``if not m.genericConverter``

**Comment by Araq** [comment]
can merge it now and nitpick later.

**Pull request merged**